source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -23,7 +23,15 @@ import * as lambda from '@aws-cdk/aws-lambda';
 import { Construct } from '@aws-cdk/core';
 
 export function buildElasticSearch(scope: Construct, domainName: string,
-  options: CfnDomainOptions, cfnDomainProps?: elasticsearch.CfnDomainProps): [elasticsearch.CfnDomain, iam.Role] {
+  options: BuildElasticSearchProps, cfnDomainProps?: elasticsearch.CfnDomainProps): [elasticsearch.CfnDomain, iam.Role] {
+
+  let zoneProps: elasticsearch.CfnDomainProps = {};
+
+  if (options.lambdaFunction) {
```

Comment:
Changed the condition to lambda but left the lambda in the prop as optional. If a lambda function is passed in, we want to use the security group ids associate with that lambda otherwise it will just use the default security group ids. Let me know what you think